src/com/cft/contactmerge/NameMatchLogic.java

```diff
@@ -0,0 +1,129 @@
+/* Created by James Page on May 8th, 2018
+ * Contains the low level business logic for matching naming properties
+ */
+
+package com.cft.contactmerge;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public abstract class NameMatchLogic {
+    /***********************************************************************
+     ****************************** Variables ******************************
+     ***********************************************************************/
+    private final static int hyphenCharLimitYesVsMaybe = 3;
+    private final static String[] namePunctuations = {".", ",", "'", "-"};
+    protected final static String[][] alternateFirstNames = // List of names to match nicknames of the same person
+            {
+                    {"Al", "Allen"},
+                    {"Al", "Alan"},
+                    {"Bob", "Robert"}
+            };
+
+    /***********************************************************************
+     ********************* Names Compare Logic Methods *********************
+     ***********************************************************************/
+    protected static boolean isNamesExactMatch(ArrayList<String> nameOne, ArrayList<String> nameTwo){
```

Comment:
I like how you make the code more readable by creating methods for small actions. However, we should have tests for every method to make sure they do (and will continue to do) what the name says they do.

Reply:
I actually had a big discussion with people on my last project about testing private/protected methods -> https://coderanch.com/t/691639/java/Implementing-Model-View-Controller-Test

Someone said if a protected/private method is a sub method of another method by the program, it doesn't need to be directly tested. And you shouldn't need to change it to public just to write a test for it. The method is already tested by the tests, that test the public method which is using this private/protected method. So the method is tested, but tested indirectly.

Reply:
I agree that it isn't necessary to test every method. Especially small/straight forward private methods. But if the method is being used by multiple classes (many of these are) or the logic isn't straight forward (some of them aren't) then there is a benefit having a unit test. It is much easier to test a method directly than indirectly.

Reply:
Okay, I will change them to public. We can discuss later in scrum maybe how I should write tests for these methods.